x/report/report.go

```diff
@@ -123,13 +123,13 @@ type RetryCollector struct {
 // Returns an error if the maximum number of retries is exceeded.
 func (c *RetryCollector) Collect(ctx context.Context, report Report) error {
 	var e *BadRequestError
-	for i := 0; i < c.maxRetry; i++ {
-		err := c.collector.Collect(ctx, report)
+	for i := 0; i < c.MaxRetry; i++ {
```

Comment:
You won't send anything if MaxRetry == 0
```suggestion
	for i := 0; i < c.MaxRetry + 1; i++ {
```